README.md

```diff
@@ -17,21 +17,66 @@ pipx install tap-googleads
 
 ## Configuration
 
-### Get refresh token
-1. GET https://accounts.google.com/o/oauth2/v2/auth?response_type=code&client_id=client_id&redirect_uri=http://127.0.0.1&scope=https://www.googleapis.com/auth/adwords&state=autoidm&access_type=offline&prompt=select_account&include_granted_scopes=true
-1. POST https://www.googleapis.com/oauth2/v4/token?code={code}&client_id={client_id}&client_secret={client_secret}&redirect_uri=http://127.0.0.1&grant_type=authorization_code
-1. POST https://www.googleapis.com/oauth2/v4/token?refresh_token={refres_token}&client_id={client_id}&client_secret={client_secret]&grant_type=refresh_token
 ### Accepted Config Options
 
 - [ ] `Developer TODO:` Provide a list of config options accepted by the tap.
 
-A full list of supported settings and capabilities for this
-tap is available by running:
+### This tap supports two sets of configs:
+
+### Using Your Own Credentials
+
+Settings required to run this tap.
+
+- `client_id` (required)
+- `client_secret` (required)
+- `developer_token` (required)
+- `refresh_token` (required)
+- `customer_login_id` (required)
+- `customer_id` (required)
+- `start_date` (optional)
 
+How to get these settings can be found in the following Google Ads documentation:
+
+https://developers.google.com/adwords/api/docs/guides/authentication
+
+If you have installed the tap you can run the following commands to see more information about the tap.
 ```bash
 tap-googleads --about
 ```
 
+### Getting A Refresh Token
+1. GET https://accounts.google.com/o/oauth2/v2/auth?response_type=code&client_id=client_id&redirect_uri=http://127.0.0.1&scope=https://www.googleapis.com/auth/adwords&state=autoidm&access_type=offline&prompt=select_account&include_granted_scopes=true
+1. POST https://www.googleapis.com/oauth2/v4/token?code={code}&client_id={client_id}&client_secret={client_secret}&redirect_uri=http://127.0.0.1&grant_type=authorization_code
+1. POST https://www.googleapis.com/oauth2/v4/token?refresh_token={refresh_token}&client_id={client_id}&client_secret={client_secret}&grant_type=refresh_token
+
+
+### Proxy OAuth Credentials
```

Comment:
Have you found this pattern elsewhere? 

My initial reaction is why should this exist in the tap? Matatika can wrap this tap, and provide the same configurations as any other connector.

Reply:
I havent found a proxy oauth pattern anywhere else. I ended up opening an issue in the meltano sdk about adding a proxy oauth authenticator in, as it also solves api token endpoints that require different headers or bodies than the normal api requests.

The issue I found when trying to add our proxy oauth to the tap was that the original tap settings were all required by default. (Made sense when there was only one way to use it). Now by checking the existence of settings in your environment, it can handle either all your own settings like before, or the proxy oauth settings. When having both auth ways in the tap, I couldn't find a solution where we could still "require" settings.

Maybe this is solved by wrapping the tap though? Are there any docs on that specifically?